sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SymbolHelper.cs

```diff
@@ -236,26 +224,39 @@ private static bool IsAnyAttributeInOverridingChain<TSymbol>(TSymbol symbol, Fun
 
         public static Accessibility GetEffectiveAccessibility(this ISymbol symbol)
         {
+            if (symbol == null)
+            {
+                return Accessibility.NotApplicable;
+            }
+
             var result = symbol.DeclaredAccessibility;
-            var currentSymbol = symbol;
+            if (result == Accessibility.Private)
+            {
+                return Accessibility.Private;
+            }
 
-            while (currentSymbol != null)
+            for (var container = symbol.ContainingType; container != null; container = container.ContainingType)
             {
-                if (currentSymbol.DeclaredAccessibility == Accessibility.Private)
-                {
-                    return Accessibility.Private;
-                }
-                if (currentSymbol.DeclaredAccessibility == Accessibility.Internal)
+                switch (container.DeclaredAccessibility)

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this 'switch' statement with 'if' statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1301)